### PR TITLE
[RFR] Migrate RadioButtonGroupInput to use useInput

### PR DESCRIPTION
**Pull request by djhi**

### CODE REVIEW

File: packages/ra-ui-materialui/src/input/RadioButtonGroupInput.js [hunk 1]
```diff
@@ -78,70 +76,41 @@ const styles = createStyles({
...
     optionText,
     optionValue,
+    resource,
+    source,
     translateChoice,
+    validate,
     ...rest
 }) => {
-    const translate = useTranslate();
+    const classes = useStyles({});
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/RadioButtonGroupInput.js]: I tend to let a way for developers to override the styles, by accepting a `classes` that I pass to `useStyles` at runtime 

See example at https://github.com/marmelab/react-admin/blob/8cdd8cedcf90967404c7e6d71a3403c3a35f8a1c/packages/ra-ui-materialui/src/button/BulkDeleteWithConfirmButton.js#L42-L54

File: packages/ra-ui-materialui/src/input/RadioButtonGroupInput.js [hunk 2]
```diff
@@ -78,70 +76,41 @@ const styles = createStyles({
...
+        resource,
+        source,
+        validate,
+        ...rest,
+    });
 
     return (
         <FormControl
             component="fieldset"
-            className={className}
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/RadioButtonGroupInput.js]: why did you remove the ability to style the main component? It's an unnecessary breaking change.

File: packages/ra-ui-materialui/src/input/RadioButtonGroupInputItem.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            label={
+                translateChoice
+                    ? translate(choiceName, { _: choiceName })
+                    : choiceName
+            }
+        />
+    );
+};
+
+export default RadioButtonGroupInputItem;
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/RadioButtonGroupInputItem.js]: the previous function used `useCallback`, this component should use `React.memo` or `useMemo` to achieve the same.

File: packages/ra-ui-materialui/src/input/RadioButtonGroupInput.spec.js [hunk 4]
```diff
@@ -1,123 +1,157 @@
...
         expect(input1.checked).toBeFalsy();
-        const input2 = getByLabelText('Female');
+        const input2 = getByLabelText('Mastercard');
         expect(input2.type).toBe('radio');
-        expect(input2.name).toBe('foo');
+        expect(input2.name).toBe('type');
         expect(input2.checked).toBeFalsy();
     });
 
     it('should use the input parameter value as the initial input value', () => {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/RadioButtonGroupInput.spec.js]: you've completely changed the tes implementation, so the test title no longer corresponds to the implementation.

File: packages/ra-ui-materialui/src/input/RadioButtonGroupInput.spec.js [hunk 5]
```diff
@@ -143,60 +182,107 @@ describe('<RadioButtonGroupInput />', () => {
...
                 />
             </TranslationContext.Provider>
         );
-        expect(queryByText('**Male**')).toBeNull();
-        expect(queryByText('Male')).not.toBeNull();
+        expect(queryByText('**Mastercard**')).toBeNull();
+        expect(queryByText('Mastercard')).not.toBeNull();
     });
 
     it('should displayed helperText if prop is present in meta', () => {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/RadioButtonGroupInput.spec.js]:
```suggestion
    it('should display helperText if prop is present in meta', () => {
```

File: packages/ra-ui-materialui/src/input/RadioButtonGroupInput.js (hunk 2 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/input/RadioButtonGroupInput.js]: I didn't remove it, it's in the `rest` prop

File: packages/ra-ui-materialui/src/input/RadioButtonGroupInputItem.js (hunk 3 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/input/RadioButtonGroupInputItem.js]: Premature optimization ?

File: packages/ra-ui-materialui/src/input/RadioButtonGroupInput.spec.js (hunk 4 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/input/RadioButtonGroupInput.spec.js]: I don't really get that comment. `initialValues` provides the initial input value and we check that is indeed applied to the underlying input

File: packages/ra-ui-materialui/src/input/RadioButtonGroupInputItem.js (hunk 3 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/input/RadioButtonGroupInputItem.js]: I mean, we used `useCallback` because we weren't using *standard* react to render the items. We should let React do its job

File: packages/ra-ui-materialui/src/input/RadioButtonGroupInput.js (hunk 1 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/input/RadioButtonGroupInput.js]: I'll probably have to check all previously migrated components for this

File: packages/ra-ui-materialui/src/input/RadioButtonGroupInput.spec.js (hunk 4 above)
Comment (fzaninotto) [reply packages/ra-ui-materialui/src/input/RadioButtonGroupInput.spec.js]: we no longer use the 'input parameter value', as the test description says
